provider/internal/queue/queue.go

```diff
@@ -127,12 +128,13 @@ func (q *Queue) worker(ctx context.Context) {
 	var (
 		c       cid.Cid
 		counter uint64
-		k       datastore.Key = datastore.Key{}
 		inBuf   deque.Deque[cid.Cid]
 	)
 
 	const baseCap = 1024
 	inBuf.SetBaseCap(baseCap)
+	k := datastore.Key{}
+	dedupCache, _ := lru.New[cid.Cid, struct{}](baseCap)
```

Comment:
Need to use maximum size of `inBuf` to deduplicate. `baseCap` is minimum size of its internal buffer.
```suggestion
	dedupCache, _ := lru.New[cid.Cid, struct{}](batchSize)
```

Reply:
I defined a specific `const` for the size of the deduplication LRU cache, it doesn't need to be as large as the maximum size of `inBuf` otherwise the memory overhead for deduplication would be significant.